exercises/state-of-tic-tac-toe/canonical-data.json

```diff
@@ -0,0 +1,373 @@
+        {
+          "uuid": "b42ed767-194c-4364-b36e-efbfb3de8788",
+          "description": "Draw",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XOX",
+              "XXO",
+              "OXO"
+            ]
+          },
+          "expected": "draw"
+        },
+        {
+          "uuid": "227a76b2-0fef-4e16-a4bd-8f9d7e4c3b13",
+          "description": "Draw",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XXO",
+              "OXX",
+              "XOO"
+            ]
+          },
+          "expected": "draw"
+        }
+      ]
+    },
+    {
+      "description": "Ongoing games",
+      "cases": [
+        {
+          "uuid": "4d93f15c-0c40-43d6-b966-418b040012a9",
+          "description": "Ongoing game",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "   ",
+              "X  ",
+              "   "
+            ]
+          },
+          "expected": "ongoing"
+        },
+        {
+          "uuid": "c407ae32-4c44-4989-b124-2890cf531f19",
+          "description": "Ongoing game",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "O  ",
+              " X ",
+              "   "
+            ]
+          },
+          "expected": "ongoing"
+        },
+        {
+          "uuid": "199b7a8d-e2b6-4526-a85e-78b416e7a8a9",
+          "description": "Ongoing game",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "X  ",
+              " XO",
+              "OX "
+            ]
+          },
+          "expected": "ongoing"
+        }
+      ]
+    },
+    {
+      "description": "Invalid boards",
+      "cases": [
+        {
+          "uuid": "1670145b-1e3d-4269-a7eb-53cd327b302e",
+          "description": "Invalid board",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XX ",
+              "   ",
+              "   "
+            ]
+          },
+          "expected": {
+            "error": "Wrong turn order: X went twice"
+          }
+        },
+        {
+          "uuid": "47c048e8-b404-4bcf-9e51-8acbb3253f3b",
+          "description": "Invalid board",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "OOX",
+              "   ",
+              "   "
+            ]
+          },
+          "expected": {
+            "error": "Wrong turn order: O started"
+          }
+        },
+        {
+          "uuid": "035a49b9-dc35-47d3-9d7c-de197161b9d4",
+          "description": "Invalid board",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XXX",
+              "XOO",
+              "XOO"
+            ]
+          },
+          "expected": {
+            "error": "Impossible board: game should have ended after X won"
+          }
```

Comment:
It is possible to reach this board from a legal game. X simply plays in the upper-left corner as the very last move. Therefore, it feels uncomfortable to call this board impossible. I understand that this is not rational play, but if this exercise is concerned with possible/impossible, then we have to consider all possible avenues through with a board can be reached, not only the rational ones.